File: wordflip-server/src/main/java/com/wordflip/dto/word/UnassignedWordsResponse.java
package com.wordflip.dto.word;

import com.wordflip.dto.common.PageMeta;

import java.util.List;

/**
 * 未入组词池响应（GET /words/unassigned）。
 */
public record UnassignedWordsResponse(
        int page,
        int size,
        long totalElements,
        int totalPages,
        List<WordSummary> words
) {

    public static UnassignedWordsResponse of(PageMeta meta, List<WordSummary> words) {
        return new UnassignedWordsResponse(meta.page(), meta.size(), meta.totalElements(), meta.totalPages(), words);
    }
}

File: wordflip-android/core-model/src/main/java/com/wordflip/core/model/group/UnassignedWordsResponse.kt
package com.wordflip.core.model.group

import com.wordflip.core.model.study.WordSummary

/** 未入组词池响应，对齐 openapi `UnassignedWordsResponse`（含 PageMeta） */
data class UnassignedWordsResponse(
    val page: Int = 0,
    val size: Int = 0,
    val totalElements: Long = 0,
    val totalPages: Int = 0,
    val words: List<WordSummary> = emptyList(),
)
